### PR TITLE
docs for re-run failed tests only

### PR DESCRIPTION
**Pull request by sebastian-lerner**

### CODE REVIEW

**Comment by sebastian-lerner** [comment]
@kbatuigas whether we want to include it in this PR or not, i dont have a strong preference but we may want to have a section to cover re-run workflow failed and re-run workflow from start in the docs. just to explain what the different re-run options are.  i dont see anything in docs for re-run workflow from failed or re-run workflow from start.

**Comment by kbatuigas** [comment]
> @kbatuigas whether we want to include it in this PR or not, i dont have a strong preference but we may want to have a section to cover re-run workflow failed and re-run workflow from start in the docs. just to explain what the different re-run options are. i dont see anything in docs for re-run workflow from failed or re-run workflow from start.

@sebastian-lerner So far I've only found a little bit of relevant content on this page https://circleci.com/docs/workflows/#rerunning-a-workflows-failed-jobs but yes we should look into adding that, perhaps in a different document.

File: jekyll/_cci2/rerun-failed-tests-only.adoc [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+```
+
+This example snippet is from a CircleCI configuration file that:
+
+. Executes Python test files that end in `.py`, 
+. Splits tests by previous timing results (you can follow xref:test-splitting-tutorial#[this tutorial] on intelligent test splitting), 
+. Stores the test results in a new directory called `test-results`, and 
+. Uploads those test results to CircleCI.  
+
+**Note:** `-o junit_family=legacy` is present to ensure that the test results being generated contain the `filename` attribute. Not included in this snippet is the key to set parallelism (read the xref:parallelism-faster-jobs#[Test splitting and parallelism] page for more information).
```
Comment (kbatuigas) [inline jekyll/_cci2/rerun-failed-tests-only.adoc:58]: Is the `-o junit_family=legacy` supposed to go somewhere in the command?

File: jekyll/_cci2/rerun-failed-tests-only.adoc [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
++
+Your `package.json` / `package-lock.json` will be updated and committed on your next push from that branch. 
+
+. Create and set up the reporter config file, if it does not already exist. In this example, the file is named `reporter-config.json`.
++
+```json
+{
+  "reporterEnabled": "spec, mocha-junit-reporter", // set the reporters
+  "reporterOptions": {
+    "mochaFile": "results/junit/junit-[hash].xml", // each suite produces its own junit :(, save them with unique hash
```
Comment (kbatuigas) [inline jekyll/_cci2/rerun-failed-tests-only.adoc:163]: Want to double check that the `:(` is supposed to be in the comment

File: jekyll/_cci2/rerun-failed-tests-only.adoc [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+       when: always
+       name: process test results (add in file path in junit)
+       command: |
+          cd ./cypress
+          node ./scripts/fix-junit.js
+     - store_test_results: 
+       path: ./cypress/results
+```  
++
+Your new testing job's `.circleci/config.yml` definition will have both snippets from steps 3 and 4 above, one right after the other.
```
Comment (kbatuigas) [inline jekyll/_cci2/rerun-failed-tests-only.adoc:197]: Do we want to say "should have both snippets" as opposed to "will have"?

**Review by kbatuigas — COMMENTED** [review]
@sebastian-lerner Just a few clarifying questions for now

File: jekyll/_cci2/rerun-failed-tests-only.adoc (hunk 1 above)
Comment (sebastian-lerner) [reply jekyll/_cci2/rerun-failed-tests-only.adoc:58]: sorry yes, the command on line 45 should be:

`pytest -o junit_family=legacy --junitxml=test-results/junit.xml $TEST_FILES`

File: jekyll/_cci2/rerun-failed-tests-only.adoc (hunk 2 above)
Comment (sebastian-lerner) [reply jekyll/_cci2/rerun-failed-tests-only.adoc:163]: you can get rid of it.

File: jekyll/_cci2/rerun-failed-tests-only.adoc [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+```json
+{
+  "reporterEnabled": "spec, mocha-junit-reporter", // set the reporters
+  "reporterOptions": {
+    "mochaFile": "results/junit/junit-[hash].xml", // each suite produces its own junit :(, save them with unique hash
+  }
+}
+```
+
+. Modify your test command to use the two reporter flags and `circleci tests run`:
```
[the remark this answers is not in the document]
Comment (sebastian-lerner) [reply jekyll/_cci2/rerun-failed-tests-only.adoc:168]: yeah that's a good idea

File: jekyll/_cci2/rerun-failed-tests-only.adoc (hunk 3 above)
Comment (sebastian-lerner) [reply jekyll/_cci2/rerun-failed-tests-only.adoc:197]: yeah also a good idea

**Pull request merged**